tools/regression_test.py

```diff
@@ -0,0 +1,772 @@
+    """Get fps and metric.
+
+    Args:
+        shell_res (int): Backend convert result: 0 is success
+        pytorch_metric (dict): Metric info of pytorch metafile
+        metric_key (str):Metric info.
+        metric_name (str): Name of metric.
+        log_path (str): Logger path.
+        metrics_eval_list (dict): Metric list from test yaml.
+        metric_info (dict): Metric info.
+
+    Returns:
+        Float: fps: FPS of the model
+        List: metric_list: metric result list
+        Bool: test_pass: If the test pass or not.
+    """
+    metric_list = []
+
+    # check if converted successes or not.
+    if shell_res != 0:
+        fps = '-'
+        metric_value = '-'
+    else:
+        # Got fps from log file
+        fps = get_info_from_log_file('FPS', log_path)
+        print(f'Got fps = {fps}')
+
+        # Got metric from log file
+        metric_value = get_info_from_log_file('metric', log_path,
+                                              metric_key)
+        print(f'Got metric = {metric_value}')
+
+    if metric_name is None:
+        logger.error(f'metrics_eval_list: {metrics_eval_list} '
+                     'has not metric name')
+    assert metric_name is not None
+
+    metric_list.append({metric_name: metric_value})
+    test_pass = calculate_metric(metric_value, metric_name,
+                                 pytorch_metric, metric_info)
+
+    # same eval_name and multi metric output in one test
+    if metric_name == 'Top 1 Accuracy':
+        metric_name = 'Top 5 Accuracy'
+        metric_value = get_info_from_log_file('metric', log_path,
+                                              'accuracy_top-5')
+        metric_list.append({metric_name: metric_value})
+        if test_pass:
+            test_pass = calculate_metric(metric_value, metric_name,
+                                         pytorch_metric, metric_info)
+
+    return fps, metric_list, test_pass
+
+
+def get_backend_fps_metric(deploy_cfg_path,
+                           model_cfg_path,
+                           convert_checkpoint_path,
+                           device_type,
+                           eval_name,
+                           logger,
+                           metrics_eval_list,
+                           pytorch_metric,
+                           metric_info,
+                           backend_name,
+                           metric_useless,
+                           convert_result,
+                           report_dict,
+                           infer_type,
+                           log_path
+                           ):
+
+    result_path = Path(convert_checkpoint_path).with_suffix('.pkl').absolute()
+
+    cmd_str = f'cd {str(Path().cwd())} && ' \
+              'python3 tools/test.py ' \
+              f'{deploy_cfg_path} ' \
+              f'{str(model_cfg_path.absolute())} ' \
+              f'--model {str(convert_checkpoint_path)} ' \
+              f'--out {str(result_path)} ' \
```

Comment:
we do not need to pass xx.pkl since it only save result to local file.

Reply:
Done